### DISCUSSION TITLE
Invalid Format: Missing 'Action Input:' after 'Action:'

### DISCUSSION THREAD

**Discussion by vansen** [crewAIInc/crewAI#1142] [Q&A]
Hi Everyone,

I need some help here, since I dont know why this behaviour is happening. I am using the sample from the readme file directly. All I adjusted, is using the Azure OpenAI Service and set this as llm for the agents.

When I start the script, I get an endless loop when it tries to call the duckduckgo_search action:

`> Entering new CrewAgentExecutor chain...
Thought: Do I need to use a tool? Yes
Action: duckduckInvalid Format: Missing 'Action Input:' after 'Action:'Do I need to use a tool? Yes
Action: duckduckgo_searchInvalid Format: Missing 'Action Input:' after 'Action:'Do I need to use a tool? Yes`

I hope this was sufficient - let me know if anymore details are needed.
What may be wrong or what are your suggestions to make this work? I have the same issue with the search action from the trip_planner example.

Thank you,
Best,
Oliver

**Comment by prodrammer** [comment]
This is the same issue as described in #103. Things seem to work with OpenAI/ChatGPT, but fail when using other models.

**Comment by tbojanowskigl** [comment]
This is defiantly sill happening, no matter what model gpt-3 or gpt-4 

Entering new ReActSingleInputOutputParser chain...
Invalid Format: Missing 'Action:' after 'Thought:[AgentStep(action=AgentAction(tool='_Exception', tool_input="Invalid Format: Missing 'Action:' after 'Thought:", log="Thought: To find out the current age of Microsoft's CEO, I first need to identify who the CEO is as of my"), observation="Invalid Format: Missing 'Action:' after 'Thought:")]
[HumanMessage(content="Invalid Format: Missing 'Action:' after 'Thought:")]

**Comment by hiddenkirby** [comment]
I'm also seeing this happen using CrewAI v0.1.32 (Python 3.11.8)

from langchain_community.llms import Ollama
mistral = Ollama(model="mistral")
... 
myagent = Agent(
llm=mistral
...

**Comment by VideoFX** [comment]
I see this nearly every time when using Ollama and mistral or openhermes.

**Comment by tyhouch** [comment]
I am getting this error and using gpt4o. I wasnt getting it when I had the agents using tools, but once I removed them I am getting this error.

**Comment by Toasterme** [comment]
got this error : 
pydantic_core._pydantic_core.ValidationError: 1 validation error for Task
expected_output
  Field required [type=missing, input_value={'description': "\nAnalyz...ick ideal destinations)}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing

**Comment by aashishpnt** [comment]
> I am getting this error and using gpt4o. I wasnt getting it when I had the agents using tools, but once I removed them I am getting this error.

any solution to this? I am also getting this for gpt-4o works completely fine in 3.5 and other model
